src/SqlVirtualMachine/SqlVirtualMachine.Test/ScenarioTests/Common.ps1

```diff
@@ -0,0 +1,250 @@
+{
+	Remove-AzResourceGroup -Name $rg.ResourceGroupName -Force
+}
+
+<#
+	.SYNOPSIS
+	Creates a virtual machine
+#>
+function Create-VM(
+	[string] $resourceGroupName, 
+	[string] $vmName,
+	[string] $location)
+{
+	$subnetName = $vmName + "subnet"
+	$vnetName= $vmName + "vnet"
+	$pipName = $vmName + "pip"
+	# Create a subnet configuration
+	$subnetConfig = New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix 192.168.1.0/24
+
+	# Create a virtual network
+	$vnet = New-AzVirtualNetwork -ResourceGroupName $resourceGroupName -Location $location `
+	   -Name $vnetName -AddressPrefix 192.168.0.0/16 -Subnet $subnetConfig
+
+	# Create a public IP address and specify a DNS name
+	$pip = New-AzPublicIpAddress -ResourceGroupName $resourceGroupName -Location $location `
+	   -AllocationMethod Static -IdleTimeoutInMinutes 4 -Name $pipName
+
+	# Rule to allow remote desktop (RDP)
+	$nsgRuleRDP = New-AzNetworkSecurityRuleConfig -Name 'RDPRule' -Protocol Tcp `
+	   -Direction Inbound -Priority 1000 -SourceAddressPrefix * -SourcePortRange * `
+	   -DestinationAddressPrefix * -DestinationPortRange 3389 -Access Allow
+
+	# Rule to allow SQL Server connections on port 1433
+	$nsgRuleSQL = New-AzNetworkSecurityRuleConfig -Name 'MSSQLRule'  -Protocol Tcp `
+	   -Direction Inbound -Priority 1001 -SourceAddressPrefix * -SourcePortRange * `
+	   -DestinationAddressPrefix * -DestinationPortRange 1433 -Access Allow
+
+	# Create the network security group
+	$nsgName = $vmName + 'nsg'
+	$nsg = New-AzNetworkSecurityGroup -ResourceGroupName $resourceGroupName `
+	   -Location $location -Name $nsgName `
+	   -SecurityRules $nsgRuleRDP,$nsgRuleSQL
+
+	$interfaceName = $vmName + 'int'
+	$interface = New-AzNetworkInterface -Name $interfaceName `
+	   -ResourceGroupName $resourceGroupName -Location $location `
+	   -SubnetId $vnet.Subnets[0].Id -PublicIpAddressId $pip.Id `
+	   -NetworkSecurityGroupId $nsg.Id
+	
+	$cred = Get-DefaultCredentialForTest
+	
+	# Create a virtual machine configuration
+	$vmConfig = New-AzVMConfig -VMName $vmName -VMSize Standard_DS13_V2 |
+	   Set-AzVMOperatingSystem -Windows -ComputerName $vmName -Credential $cred -ProvisionVMAgent -EnableAutoUpdate |
+	   Set-AzVMSourceImage -PublisherName 'MicrosoftSQLServer' -Offer 'SQL2017-WS2016' -Skus 'Enterprise' -Version 'latest' |
+	   Add-AzVMNetworkInterface -Id $interface.Id
+	
+	return New-AzVM -ResourceGroupName $resourceGroupName -Location $location -VM $vmConfig
+}
+
+<#
+	.SYNOPSIS
+	Creates default credentials for testing
+#>
+function Get-DefaultCredentialForTest()
+{
+	$user = Get-DefaultUser
+	$pswd = Get-DefaultPassword
+	$securePswd = ConvertTo-SecureString -String $pswd -AsPlainText -Force
+	return New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $user, $securePswd
+}
+
+function Get-LocationForTest()
+{
+	return 'eastus'
+}
+
+function Get-ResourceGroupNameForTest()
+{
+	$nr = Get-Random -Minimum 1000 -Maximum 5000
```

Comment:
So, I think this is the source of the test problems.  You need to use `getAssetName``` so that the value you use in recording will be reused in playback.